plugin/storage/grpc/shared/interface.go

```diff
@@ -46,28 +48,61 @@ type StoragePlugin interface {
 	DependencyReader() dependencystore.Reader
 }
 
+// ArchiveStoragePlugin is the interface we're exposing as a plugin.
+type ArchiveStoragePlugin interface {
+	ArchiveSpanReader() ArchiveReader
+	ArchiveSpanWriter() ArchiveWriter
+}
+
+// ArchiveReader finds and loads traces and other data from storage.
+type ArchiveReader interface {
```

Comment:
is it necessary to introduce ArchiveReader/ArchiveWriter interfaces? Can we not use just spanstore.Reader/spanstore.Writer?

Reply:
I introduce ArchiveReader/ArchiveWriter to have separate method names from spanstore.Reader/spanstore.Writer implemented in grpc plugin. I believe it is necessary to distinguish archive read/write or not.